File: app/schemas/vision.py
"""视觉预识别响应 Schema（对齐 app/routers/vision.py）。"""
from __future__ import annotations

from pydantic import BaseModel


class VisionCategory(BaseModel):
    """可选分类（供前端手动改类下拉）。"""

    id: int
    name: str


class VisionPredictResponse(BaseModel):
    """视觉预识别结果。

    字段与 `VisionService.predict` 的契约一致：
    `category_id` / `label` / `confidence`（加 `categories` 供前端手动纠偏）。
    """

    category_id: int
    label: str
    confidence: float
    categories: list[VisionCategory] = []

File: web/src/api/vision.ts
// 视觉预识别接口（对齐 app/routers/vision.py）
// 发布前上传首图，返回 AI 识别结果（category_id / label / confidence）及可选分类列表。
import { apiPost } from './request'
import { compressImage } from '@/utils/image'
import type { VisionPredictResult } from '@/types'

// 识别 = 上传 + 后端 CPU 推理，公网下慢；给足 60s（全局 axios 默认 15s 不够）。
const UPLOAD_TIMEOUT = 60000

export const visionApi = {
  async predict(file: File): Promise<VisionPredictResult> {
    // 先压缩手机原图（几 MB → 几百 KB），上传与后端推理都快数倍
    const small = await compressImage(file)
    const fd = new FormData()
    fd.append('image', small)
    return apiPost<VisionPredictResult>('/vision/predict', fd, { timeout: UPLOAD_TIMEOUT })
  },
}
